tabular/src/autogluon/tabular/learner/default_learner.py

```diff
@@ -187,6 +192,17 @@ def general_data_processing(self, X: DataFrame, X_val: DataFrame, X_unlabeled: D
             if X_unlabeled is not None:
                 X_unlabeled = X_super.tail(len(X_unlabeled)).set_index(X_unlabeled.index)
             del X_super
+        if w is not None:
+            X[self.weight_column] = w
+            if X_val is not None:
+                if w_val is not None:
+                    X_val[self.weight_column] = w_val
+                elif not self.weight_evaluation:
+                    nan_vals = np.empty((len(X_val),))
+                    nan_vals[:] = np.nan
+                    X_val[self.weight_column] = nan_vals
+                else:
+                    raise ValueError(f"Weight column '{self.weight_column}' cannot be missing from X_val with weight_evaluation specified.")
 
         return X, y, X_val, y_val, X_unlabeled, holdout_frac, num_bag_folds
```

Comment:
I feel that this should instead return w, w_val instead of keeping in X. We don't want trainer knowing anything about the weight column name, in the same way trainer doesn't know about the label column name.

Reply:
Will keep current internal implementation with weights bundled inside X for now as discussed. Future contributors may otherwise not remember to handle sample-weights when they are adding new functionality, so alternative may need careful design. Added TODO for this issue in Trainer.